tests/integration/payments.test.js

```diff
@@ -232,56 +212,51 @@ describe('PAYMENTS ROUTES - POST /payments/createlist', () => {
         headers: { Cookie: `alenvi_token=${authToken}` },
       });
 
-      expect(response.statusCode).toEqual(500);
-
-      const paymentCountAfter = await Payment.countDocuments({ company: authCompany._id });
-      expect(paymentCountAfter).toEqual(paymentCountBefore);
-      const paymentNumberAfter = await PaymentNumber.findOne({
-        prefix: '0319',
-        nature: 'payment',
-        company: authCompany._id,
-      }).lean();
-      expect(paymentNumberBefore.seq).toEqual(paymentNumberAfter.seq);
-      sinon.assert.notCalled(generateXML);
-      formatPaymentNumber.restore();
-      generateXML.restore();
+      expect(response.statusCode).toBe(404);
     });
 
-    it('should not create multiple payments if at least one customer is not from the same company', async () => {
-      const payload = [
-        { ...originalPayload[0], customer: customerFromOtherCompany._id },
-        { ...originalPayload[1] },
-      ];
+    it('should return a 400 if user tries to create a payment with an existing number', async () => {
+      const generateXML = sinon.stub(PaymentHelper, 'generateXML');
+
+      const payload = [{ ...originalPayload[0], number: paymentsList[0].number }];
+
       const response = await app.inject({
         method: 'POST',
         url: '/payments/createlist',
         payload,
         headers: { Cookie: `alenvi_token=${authToken}` },
       });
-      expect(response.statusCode).toBe(403);
+
+      expect(response.statusCode).toEqual(400);
+      sinon.assert.notCalled(generateXML);
+      generateXML.restore();
     });
 
-    it('should not create multiple payments if at least one paiement has a tpp', async () => {
+    it('should return a 400 if at least one paiement has a tpp', async () => {
       const payload = [
         { ...originalPayload[0], thirdPartyPayer: new ObjectID() },
         { ...originalPayload[1] },
       ];
+
       const response = await app.inject({
         method: 'POST',
         url: '/payments/createlist',
         payload,
         headers: { Cookie: `alenvi_token=${authToken}` },
       });
+
       expect(response.statusCode).toBe(400);
     });
   });
 
-  describe('Admin without company', () => {
+  describe('CLIENT_ADMIN without company credentials', () => {
```

Comment:
A-t-on besoin de ce test ? J'ai l'impression que c'est un test du service, non ?

Reply:
Je ne suis pas sure de moi mais est-ce que c'est parce que dans la methode `CreateList` dans paymentController on met les credentials dans l'appel: 
`const payments = await PaymentHelper.savePayments(req.payload, req.auth.credentials);` ?

Reply:
Je ne crois pas, le `req.auth.credentials` est toujours defini si la personne est connecté car on le definie dans `authorization`. 
Je me demande si ce n'est pas lié au premier if de `savePayments`

Reply:
Ah oui en effet. Comme c'est teste unitairement, (`should return error if company is missing'`)  je peux le retirer tu as raison !